validator/src/main.rs

```diff
@@ -2575,8 +2611,15 @@ pub fn main() {
                     let shred_storage_size =
                         value_t_or_exit!(matches, "rocksdb_fifo_shred_storage_size", u64);
                     ShredStorageType::RocksFifo(BlockstoreRocksFifoOptions {
-                        shred_data_cf_size: shred_storage_size / 2,
-                        shred_code_cf_size: shred_storage_size / 2,
+                        // Allocate the size between data/coding shreds based
+                        // on their size ratio that also take compression into
+                        // consideration.
+                        shred_data_cf_size: shred_storage_size
+                            / (data_shred_size_ratio + code_shred_size_ratio)
+                            * data_shred_size_ratio,
+                        shred_code_cf_size: shred_storage_size
+                            / (data_shred_size_ratio + code_shred_size_ratio)
+                            * code_shred_size_ratio,
```

Comment:
The size allocation code is updated for both data shreds and coding shreds.  Note that the division goes before multiplication to avoid the possible overflow (although very unlikely).

Reply:
nit: Think we can simplify by multiplying by values like `0.5` here

Reply:
I can revert this part.  Previously I was trying to dynamically adapt the ratio based on whether we compress ShredData.  But since we are not compressing ShredData at this moment, we can revert back to the `/2` approach.

Reply:
Sure; divide-by-2 works or multiplying by a float accomplishes the same thing and allows us to tweak ratios easier